app/kuma-dp/pkg/dataplane/metrics/server.go

```diff
@@ -58,16 +60,36 @@ type (
 	OtelMutator    func(in io.Reader) ([]*io_prometheus_client.MetricFamily, error)
 )
 
-type QueryParametersModifier func(queryParameters url.Values) string
+type QueryParametersModifier func(queryParameters url.Values) url.Values
 
-func RemoveQueryParameters(_ url.Values) string {
-	return ""
+func RemoveQueryParameters(_ url.Values) url.Values {
+	return url.Values{}
 }
 
-func AddPrometheusFormat(queryParameters url.Values) string {
+func AddPrometheusFormat(queryParameters url.Values) url.Values {
 	queryParameters.Add("format", "prometheus")
 	queryParameters.Add("text_readouts", "")
-	return queryParameters.Encode()
+	return queryParameters
+}
+
+func AddUsedOnlyParameter(sidecar *v1alpha12.Sidecar) func(queryParameters url.Values) url.Values {
+	values := v1alpha1.EnvoyMetricsFilter(sidecar)
+
+	return func (queryParameters url.Values) url.Values {
```

Comment:
waiting to confirm https://github.com/kumahq/kuma/blob/12ab22c7673ac76d6e8b84ddd46bd647888ad56b/pkg/xds/generator/prometheus_endpoint_generator.go#L191 if it's still the case, doesn't look like it

```
# krzysztof.slonka@konghq.com @ K-Sonka-MBP in ~/projects/kuma on git:meshmetric-usedonly-filters x(⎈|kong-cloud-02-dev-eu-west-1:N/A) [9:07:39]
$ curl -s 'http://localhost:9901/stats?usedonly=' | wc -l
      79

# krzysztof.slonka@konghq.com @ K-Sonka-MBP in ~/projects/kuma on git:meshmetric-usedonly-filters x(⎈|kong-cloud-02-dev-eu-west-1:N/A) [9:07:44]
$ curl -s 'http://localhost:9901/stats?' | wc -l
     437
```

Reply:
@lukidzi says it may no longer be the case - do you think I should go and update other places in the codebase to use one function?

Reply:
I was playing with it recently, and it worked, so this comment is outdated